python-package/xgboost/core.py

```diff
@@ -313,86 +313,138 @@ def _prediction_output(shape, dims, predts, is_cuda):
     return arr_predict
 
 
-class DataIter:
-    '''The interface for user defined data iterator. Currently is only supported by Device
-    DMatrix.
+class DataIter:  # pylint: disable=too-many-instance-attributes
+    """The interface for user defined data iterator.
+
+    Parameters
+    ----------
+    cache_prefix:
+        Prefix to the cache files, only used in external memory.  It can be either an URI
+        or a file path.
+
+    """
+    _T = TypeVar("_T")
+
+    def __init__(self, cache_prefix: Optional[str] = None) -> None:
```

Comment:
`cache_prefix` can be a parameter for `DMatrix` instead of `DataIter`.  I don't have a strong preference for the choice.  But do note that it's useful since users might have a URI that's not a local file path so we can't drop the parameter.